packages/synapse-react-client/src/lib/utils/hooks/SynapseAPI/dataaccess/useAccessRequirements.ts

```diff
@@ -129,3 +148,181 @@ export function useCreateLockAccessRequirement(
+          return options.onSuccess(data, variables, ctx)
+        }
+      },
+    },
+  )
+}
+
+export function useGetDataAccessRequestForUpdate(
+  accessRequirementId: string,
+  options?: UseQueryOptions<Request | Renewal, SynapseClientError>,
+) {
+  const { accessToken, keyFactory } = useSynapseContext()
+
+  return useQuery<Request | Renewal, SynapseClientError>(
+    keyFactory.getDataAccessRequestForUpdateQueryKey(accessRequirementId),
+    () =>
+      SynapseClient.getDataAccessRequestForUpdate(
+        accessRequirementId,
+        accessToken!,
+      ),
+    options,
+  )
+}
+
+export function useUpdateDataAccessRequest(
+  options?: UseMutationOptions<
+    Request | Renewal,
+    SynapseClientError,
+    Request | Renewal
+  >,
+) {
+  const { accessToken } = useSynapseContext()
+  const queryClient = useQueryClient()
+  const { keyFactory } = useSynapseContext()
+
+  return useMutation<Request | Renewal, SynapseClientError, Request | Renewal>(
+    (requestInterface: Request | Renewal) =>
+      SynapseClient.updateDataAccessRequest(requestInterface, accessToken!),
+    {
+      ...options,
+      onSuccess: async (data, variables, ctx) => {
+        // Invalidate the data access request query
+        await queryClient.invalidateQueries(
+          keyFactory.getDataAccessRequestForUpdateQueryKey(
+            data.accessRequirementId,
+          ),
+        )
+        if (options?.onSuccess) {
+          return options.onSuccess(data, variables, ctx)
+        }
+      },
+    },
+  )
+}
+
+export function useCreateAccessApproval(
+  options?: UseMutationOptions<
+    AccessApproval,
+    SynapseClientError,
+    AccessApproval
+  >,
+) {
+  const { accessToken } = useSynapseContext()
+  const queryClient = useQueryClient()
+  const { keyFactory } = useSynapseContext()
+
+  return useMutation<AccessApproval, SynapseClientError, AccessApproval>(
+    request => SynapseClient.createAccessApproval(accessToken, request),
+    {
+      ...options,
+      onSuccess: async (data, variables, ctx) => {
+        // Invalidate query for AR status
+        await queryClient.invalidateQueries(
+          keyFactory.getAccessRequirementStatusQueryKey(
+            String(variables.requirementId),
+          ),
+        )
+        if (options?.onSuccess) {
+          return options.onSuccess(data, variables, ctx)
+        }
+      },
+    },
+  )
+}
+
+export function useCancelDataAccessRequest(
+  options?: UseMutationOptions<
+    ACTSubmissionStatus,
+    SynapseClientError,
+    { submissionId: string; accessRequirementId: string }
+  >,
+) {
+  const { accessToken } = useSynapseContext()
+  const queryClient = useQueryClient()
+  const { keyFactory } = useSynapseContext()
+
+  return useMutation<
+    ACTSubmissionStatus,
+    SynapseClientError,
+    { submissionId: string; accessRequirementId: string }
+  >(
+    request =>
+      SynapseClient.cancelDataAccessRequest(request.submissionId, accessToken!),
```

Comment:
We don't need the the `accessRequirementId` to cancel the request, but we do need it to invalidate the `AccessRequirementStatus` query key, so we require it as an argument.